lib/bolt/plugin/module.rb

```diff
@@ -150,15 +160,29 @@ def process_params(task, opts)
         # handled previously. That may not always be the case so filter them
         # out now.
         _meta, params = opts.partition { |key, _val| key.start_with?('_') }.map(&:to_h)
-
         metaparams = {}
-        metaparams['_config'] = config if config?
+        # Send config with `_config` when config is defined in bolt_plugin.json
+        # Otherwise, merge config with params
+        # TODO: remove @send_config when deprecated
+        if @send_config
+          metaparams['_config'] = config if config?
+        else
+          params = @config ? config.merge(params) : params
+        end
         metaparams['_boltdir'] = @context.boltdir
 
         validate_params(task, params)
         [params, metaparams]
       end
 
+      def extract_task_parameter_schema
+        @hook_map.each_with_object({}) do |(_hook, task), acc|
+          if (schema = task['task'].metadata['parameters'])
+            acc.merge!(schema)
```

Comment:
This needs to be the *intersection* of types. Also, while i'm here I might as well wrap in `Optional`.